src/test/java/Pages/DashboardPage.java

```diff
@@ -1,63 +1,284 @@
+        assertTrue(activitiesModalXPathLocator.isDisplayed());
+    }
+
+    public void closeActivityModal() {
+        cancelButtonXPathLocator.click();
+    }
+
+    public List<String> fillActivity(String projectOptionValue, String categoryOptionValue, String hourValue, String ticketValue, String commentsValue) {
+        List<String> activityValues = new ArrayList<String>();
+        WebElement today = getTodayElement();
+
+        activityValues.add(today.getText());
+        activityValues.add(ticketValue);
+
+        projectComboboxXPathLocator.click();
+        WebElement projectOption = getProjectOptionElement(projectOptionValue);
+        wait.until(ExpectedConditions.visibilityOf(projectOption));
+        projectOption.click();
+
+        categoriesComboboxXPathLocator.click();
+        WebElement categoryOption = getCategoryOptionElement(categoryOptionValue);
+        wait.until(ExpectedConditions.visibilityOf(categoryOption));
+        categoryOption.click();
+
+        WebElement hourInput = getHourElement();
+        hourInput.clear();
+        hourInput.sendKeys(hourValue);
+
+        WebElement ticketInput = getTicketElement();
+        ticketInput.clear();
+        ticketInput.sendKeys(ticketValue);
+
+        WebElement commentsTextarea = getCommentsElement();
+        commentsTextarea.clear();
+        commentsTextarea.sendKeys(commentsValue);
+
+        createButtonXPathLocator.click();
+
+        wait.until(ExpectedConditions.visibilityOf(successfulSnackbarXPathLocator));
+
+        if (successfulSnackbarXPathLocator.getText().contains("created")) {
+            assertEquals(successfulSnackbarXPathLocator.getText(), "Activity successfully created");
+        } else {
+            assertEquals(successfulSnackbarXPathLocator.getText(), "Activity successfully updated");
+        }
+
+        return activityValues;
+    }
+
+    public void clickOnSelectedActivity(String today, String ticketValue) {
+        String day;
+        int indexDateHeader = 0, indexTicketValue = 0;
+
+        for (int i = 0; i < calendarHeaderXPathLocator.size(); i++) {
+            day = calendarHeaderXPathLocator.get(i).findElement(By.xpath("./div/div")).getText();
+
+            if (day.equals(today)) {
+                indexDateHeader = i;
+            }
+        }
+
+        List<WebElement> todayColumnActivities = calendarBodyXPathLocator.get(indexDateHeader).findElements(By.xpath("./div/div"));
+
+        for (int j = 0; j < todayColumnActivities.size(); j++) {
+            if (todayColumnActivities.get(j).findElement(By.xpath("./div/p/following-sibling::span[1]")).getText().equals(ticketValue)) {
+                indexTicketValue = j;
+            }
+        }
+
+        WebElement todaySelectedActivity = todayColumnActivities.get(indexTicketValue);
+        todaySelectedActivity.click();
+    }
+
+    public void clickOnSelectedActivityToDelete(String today, String ticketValue) throws InterruptedException {
```

Comment:
Code from this method and the one above is pretty similar, maybe you could modify clickOnSelectedActivity to getSelectedActivity and leave the click on the element out for update/validate and then navigate in it to click on the ellipsis menu to delete/clone.